### ISSUE TITLE
`cargo clippy` finds `warning: passing a unit value to a function` for certain types (deriving `Archive`).

### ISSUE THREAD

**Issue by DarrenTsung** [djkoloski/rkyv#90]
I made a repro repo here: https://github.com/DarrenTsung/rkyv-clippy-repro.

I'm seeing a weird issue where running `cargo clippy` on certain types outputs a warning with some potentially weird generated code? The minimal repro is:

```rust
use rkyv::*;

#[derive(Archive, Deserialize, Serialize)]
pub struct Foo(u32);

#[derive(Archive, Deserialize, Serialize)]
pub struct Bar {
    id: u32,
}
```

When I run `cargo clippy` I get:
```
❯ cargo clippy
warning: passing a unit value to a function
 --> src/lib.rs:4:16
  |
4 | pub struct Foo(u32);
  |                ^^^
  |
  = note: `#[warn(clippy::unit_arg)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg
help: move the expression in front of the call and replace it with the unit literal `()`
  |
4 | pub struct Foo({
5 |     Archive;
6 |     u32
7 | });
  |

warning: passing a unit value to a function
 --> src/lib.rs:8:5
  |
8 |     id: u32,
  |     ^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg
help: move the expression in front of the call and replace it with the unit literal `()`
  |
8 |     {
9 |         id;
10|         ()
11|     }: u32,
  |

warning: 2 warnings emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
```

My clippy version is:
```
❯ cargo clippy --version
clippy 0.0.212 (cb75ad5d 2021-02-10)
```

Any thoughts why this might be happning?

**Comment by DarrenTsung** [comment]
Oh, also I took at the expanded code with `cargo expand` and couldn't find any issues that looked like the samples clippy reported.

**Comment by djkoloski** [comment]
I made sure my clippy version matched yours:

```
> cargo clippy --version
clippy 0.0.212 (cb75ad5d 2021-02-10)
```

But I don't get any warnings when I run `cargo clippy` or `cargo clippy -- -W clippy::pedantic` with your test repo. Do you have a custom clippy config?

**Comment by djkoloski** [comment]
Darren helped work through some of this on Discord. Basically, clippy was exhibiting strange behavior for me and was linting with `cargo clippy`, but only when `rkyv` wasn't a path dependency. Running `cargo +nightly clippy -Zunstable-options` got a successful repro for path dependencies, and I used that to figure out where to put a few `#[allow(clippy::unit_arg)]`s.

Fixed by b4e757d6e6fad5b8c8d4fcbec55b67cb5c637784.
